website/static/js/pages/file-page.js

```diff
@@ -58,4 +58,42 @@ $(function() {
         });
     }
 
+    if(window.contextVars.currentUser.canEdit && !window.contextVars.node.isRegistration) {
```

Comment:
At a minimum this should handle checkouts-- iirc disabling the button if there is a `window.contextVars.file.checkoutUser` would stop bad things from happening.

Also, multiple addons do not allow renames (iirc: dataverse, maybe published figshares, bitbucket is read-only). You should be able to access context vars `window.contextVars.file.provider` to disable the button for those providers easily enough.

Reply:
For this ticket, we're going to let the API fail, but there's a follow-up ticket to more gracefully handle these things in the future. https://openscience.atlassian.net/browse/OSF-8776